### PR TITLE
Ads mercury fix

### PR DESCRIPTION
**Pull request by emadry**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 1]
```diff
@@ -496,6 +496,38 @@ public boolean checkSlotOnPageLoaded(String slotName) {
...
+      }
+
+      List<WebElement> adWebElements = slot.findElements(By.cssSelector("iframe"));
+
+      Log.log("Slot found",
+              String.format("%s found on the page with selector: %s", slotName, slotSelector),
+              true
+      );
+
+      return adWebElements.size() >= 1;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)

File: src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestUap.java [hunk 2]
```diff
@@ -24,15 +24,16 @@
...
   ) {
     AdsBaseObject ads = new AdsBaseObject(driver, page.getUrl(RESOLVED_STATE), WindowSize.DESKTOP);
-    verifySlotsUnblocked(ads, atfSlots);
+    verifySlotsUnblocked(ads, atfSlots, false);
     verifySlotsBlocked(ads, btfSlots);
-    verifySlotsUnblocked(ads, ListUtils.union(atfSlots, btfSlots));
+    verifySlotsUnblocked(ads, ListUtils.union(atfSlots, btfSlots), false);
   }
 
   @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5_DEFAULT // FIXME: use default emulator if mobile flag will be reverted
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestUap.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1134)
MINOR Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L229](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L229): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L230](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L230): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L494](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L494): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L572](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L572): Remove this unused private "waitForProvidersChain" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L730](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L730): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L759](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L759): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L1006](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L1006): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [AdsBaseObject.java#L46](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L46): Define a constant instead of duplicating this literal "data-gpt-line-item-id" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [TestAdsBtfBlocking.java#L22](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestAdsBtfBlocking.java#L22): Define a constant instead of duplicating this literal "project43" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [TestAdsBtfBlocking.java#L59](https://github.com/Wikia/selenium-tests/blob/f0a86a6e0641a2cd9445966db70bd9875a5867ed/src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestAdsBtfBlocking.java#L59): Define a constant instead of duplicating this literal "Slot: " 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L230](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L230): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L231](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L231): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L496](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L496): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L574](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L574): Remove this unused private "waitForProvidersChain" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L732](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L732): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L761](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L761): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L1008](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L1008): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [AdsBaseObject.java#L46](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L46): Define a constant instead of duplicating this literal "data-gpt-line-item-id" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [TestAdsBtfBlocking.java#L22](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestAdsBtfBlocking.java#L22): Define a constant instead of duplicating this literal "project43" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [TestAdsBtfBlocking.java#L59](https://github.com/Wikia/selenium-tests/blob/fced6d8496ca19fc1718cf04e30be8ff944cfdeb/src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestAdsBtfBlocking.java#L59): Define a constant instead of duplicating this literal "Slot: " 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 3]
```diff
@@ -496,6 +498,38 @@ public boolean checkSlotOnPageLoaded(String slotName) {
       restoreDefaultImplicitWait();
     }
   }
+  /**
+   * Check if AdEngine loaded the ad web elements inside slot with mobile state
+   */
+  public boolean checkSlotOnPageLoaded(String slotName, Boolean isMobile) {
```
Comment (Bielik20) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: Consider making base method (without Boolean param) to call this one. This way you would eliminate code duplication.

**Pull request merged**